src/EditorFeatures/Core/EditAndContinue/EditAndContinueLanguageService.cs

```diff
@@ -245,24 +254,34 @@ private ActiveStatementSpanProvider GetActiveStatementSpanProvider(Solution solu
 
         /// <summary>
         /// Returns true if any changes have been made to the source since the last changes had been applied.
+        /// For performance reasons it only implements a heuristic and may return both false positives and false negatives.
+        /// If the result is a false negative the debugger will not apply the changes unless the user explicitly triggers apply change command.
+        /// The background diagnostic analysis will still report rude edits for these ignored changes. It may also happen that these rude edits 
+        /// will disappear once the debuggee is resumed - if they are caused by presence of active statements around the change.
+        /// If the result is a false positive the debugger attempts to apply the changes, which will result in a delay but will correctly end up
+        /// with no actual deltas to be applied.
         /// </summary>
-        public async ValueTask<bool> HasChangesAsync(string? sourceFilePath, CancellationToken cancellationToken)
+        public ValueTask<bool> HasChangesAsync(string? sourceFilePath, CancellationToken cancellationToken)
```

Comment:
I assume I need to call those with .ConfigureAwait(false) if calling from a JTF.Run() from the main thread?

Reply:
It has not changed in that regard - we use `ConfigureAwait(false)` by default in Roslyn, so `EditSession.HasChangesAsync` does as well.